instrumentation/testing/benchmark.go

```diff
@@ -119,6 +120,17 @@ func startBenchmark(b *testing.B, pc uintptr, benchFunc func(b *testing.B)) {
 		funcName = fullTestName[:testNameSlash]
 	}
 	packageName := getBenchmarkSuiteName(b)
+	if packageName == "" {
+		funcFullName := runtime.FuncForPC(pc).Name()
+		funcNameIndex := strings.LastIndex(funcFullName, funcName)
+		if funcNameIndex < 1 {
+			funcNameIndex = len(funcFullName)
+		}
+		packageName = funcFullName[:funcNameIndex-1]
+		if len(packageName) > 0 && packageName[0] == '_' && strings.Index(packageName, sourceRoot) != -1 {
```

Comment:
Can we extract this logic to some common function? It seems it's the same than `testing.go` file. 
Furthermore, I'd add some comments to explain what it is doing here.

Reply:
moved to `util.go` file

Reply:
also commented.